finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/service/H2ReqRep.scala

```diff
@@ -0,0 +1,66 @@
+package com.twitter.finagle.buoyant.h2
+package service
+
+import com.twitter.finagle.buoyant.h2.service.H2ReqRep.{FinalFrame, RepAndFrame}
+import com.twitter.finagle.service.ReqRep
+import com.twitter.util.{Return, Throw, Try}
+
+/**
+ * Like a [[com.twitter.finagle.service.ReqRep]], but for H2
+ *
+ * @param request  a H2 [[Request]]
+ * @param response a [[Try]] containing either the H2 [[Response]] and the
+ *                 final [[Frame]], or an error. the final frame is contained in
+ *                 both an [[Option]], so that empty streams can pass [[None]], and
+ *                 in a [[Try]], in case the [[Stream]] [[com.twitter.util.Throw Throw]]s.
+ */
+class H2ReqRep(override val request: Request, override val response: Try[RepAndFrame])
+  extends ReqRep(request, response) {
+
+  /**
+   * the [[Response]] part of `this.response`,
+   * ignoring the final frame of the response stream
+   */
+  def responseOnly: Try[Response] = response.map { case (rsp, _) => rsp }
```

Comment:
@olix0r I (very briefly) had these methods as `lazy val`s instead of `def`s, but I figured it would make Alex mad. Since this is in the hot path, and the parameters are `val`s, can I _pleeeease_? :)